contracts/SpokePool.sol

```diff
@@ -146,20 +174,65 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
         numberOfDeposits += 1;
     }
 
-    function initiateRelay(
-        uint256 originChain,
+    function fillRelay(
         address sender,
-        uint256 amount,
         address recipient,
-        uint256 relayerFee,
-        uint256 realizedLpFee
-    ) public {}
-
-    function fillRelay(
-        uint256 relayId,
+        address destinationToken,
+        uint64 realizedLpFeePct,
+        uint64 relayerFeePct,
+        uint64 depositId,
+        uint256 originChainId,
+        uint256 relayAmount,
```

Comment:
When we say "fillAmount", is this the amount of tokens that are transferred from the relayers wallet? Or something else? What do you think the right UX for this is?

We could make this the max amount they're willing to have pulled from their wallet?

Thoughts?

Reply:
it's not the amount pulled from the Wallet. it's the amount pulled plus the realised LP Fee, plus the relayer fee. put another way, the amount sent from the relayer(and thus pulled from their wallet) is `fillAmount * (1 - relayerFeePct - realizedLpFeePct`

Reply:
Yeah chris is correct

Reply:
So a few points on this:
1. I think it's a bit unclear what fillAmount is, although we might be able to resolve this via comments.
2. If a relayer is capital limited (which is the primary motivation for partial fills), they would likely want to use this method to send exactly as many tokens as they have to the contract. Instead of requiring them to compute fillAmount from that, why can't we just let them specify how many tokens they want to send (or the max amount they are willing to send) as that's what they would care about I would think?

Reply:
ye, I agree with that 2nd point. it's confusing. what should we call it though @mrice32 ?

Reply:
what if we just renamed them to `relayAmount => totalAmountToRelay` and `fillAmount => amountToRelay`...?

Reply:
I don't quite understand point 2 @mrice32

Reply:
Sorry, I think I'm making this too confusing. That's my bad.

If I have 1000 USDC, it's difficult to tell this method that I only want to send 1000 USDC since you have to do some math to get there. Instead, you could have a field called `maxTokensToSend` or similar, that allows the caller to specify the max number of tokens they're willing to contribute.

Not only is the easier for the caller, but it's nice from a frontrunning perspective too. If I try to fill 1000 USDC, but someone frontruns my txn and there's only room for 500 USDC, my transaction still goes through, it just pulls less from my wallet.

Reply:
the point is that `fillAmount`, the thing that the relayer is passing into the prop, is not the actual amount of funds that gets pulled from their wallet. I.e if the relay is 100 weth but I only have 5 weth I cant put `fillAmount` to 5 weth as this won't use _all_ my funds, it'll pull `5*(1-relayerFeePct-realizedLpFeePct)` thereby leaving over some in my wallet. If I want to spend the _full_ 5 WETH I will need some compute `5*(1+relayerFeePct+realizedLpFeePct)` thereby pulling exactly 5 WETH from my wallet.

The point here is the prop name `fillAmount` is a bit confusing as ideally the amount the relayer chooses for this prop is what is pulled from their wallet so they can choose exactly how much to send.

Reply:
ah i see so is it naive to just change the name to `maxAmountToFill`? or `fillAmountPreFees`? This way the caller has more clarity that they won't send the full amount that they pass in?

Reply:
I think that's ok for now @nicholaspai . we might want to re-consider this to be the amount that's actually pulled from their wallet but that's a bit more complex to reason about. a good name could be `fillAmountSansFees` or like you have it with `PreFees`.

Reply:
Hmmm, I don't really think the name is the core issue. I think it's the fact that a caller has no way to say "I want to send this amount of funds". Is it easy to compute fillAmount from maxTokensToSend in this function?

Reply:
it's trivial to compute this I think. if we shifted it to this you can modify some of the logic within this function to deal with this. if the prop was called `relayerFillAmount` or `relayerAmountSpend` indicating how much funds are actually pulled from the relayers wallet, then you would fill the amount of the original relay by `relayFillAmount*(1+relayerFeePct+realizedLpFeePct)`. i.e the amount _filled_ includes these fees but the amount _sent_ is before the fees. 

it's really just a framing question and both are equally simple to implement IMHO.

Reply:
I think I understand what you're saying. In this [commit](https://github.com/across-protocol/across-smart-contracts-v2/pull/9/commits/977ffa9d7db0e6f590c44d15f8a963e80d0737e3) I allow the user to pass in `maxTokensToSend`, and the contract now computes internal accounting of `fillAmountPreFees`